stress/v2/statement/set.go

```diff
@@ -0,0 +1,87 @@
+package statement
+
+import (
+	"fmt"
+	"strings"
+
+	influx "github.com/influxdata/influxdb/client/v2"
+	"github.com/influxdata/influxdb/stress/v2/ponyExpress"
+)
+
+// SetStatement set state variables for the test
+type SetStatement struct {
+	Var   string
+	Value string
+
+	StatementID string
+
+	Tracer *ponyExpress.Tracer
+}
+
+// SetID statisfies the Statement Interface
+func (i *SetStatement) SetID(s string) {
+	i.StatementID = s
+}
+
+// Run statisfies the Statement Interface
+func (i *SetStatement) Run(s *ponyExpress.StoreFront) {
+
+	// Set the Tracer
+	i.Tracer = ponyExpress.NewTracer(make(map[string]string))
+
+	// Create a new Directive
+	d := ponyExpress.NewDirective(strings.ToLower(i.Var), strings.ToLower(i.Value), i.Tracer)
+
+	switch d.Property {
+
+	// Needs to be set on both StoreFront and ponyExpress
+	// Set the write percison for points generated
+	case "precision":
+		s.Precision = d.Value
+
+		// Increment the tracer
+		i.Tracer.Add(1)
+		s.SendDirective(d)
+
+	// Lives on StoreFront
+	// Set the date for the first point entered into the database
+	case "startdate":
+		s.Lock()
+		defer s.Unlock()
```

Comment:
These defers are going to "leak outside the switch" and not get evaluated until after `i.Tracer.Wait()`. Not sure if that's intended or correct.